ui/pages/send/send-content/send-content.component.js

```diff
@@ -77,11 +121,83 @@ export default class SendContent extends Component {
       recipient.warning === 'knownAddressRecipient';
     const hideAddContactDialog = recipient.warning === 'loading';
 
+    let title;
+    if (
+      draftTransaction?.asset.details?.standard === ERC721 ||
+      draftTransaction?.asset.details?.standard === ERC1155
+    ) {
+      title = draftTransaction?.asset.details?.name;
+    } else if (draftTransaction?.asset.details?.standard === ERC20) {
+      title = `${hexWEIToDecETH(draftTransaction.amount.value)} ${
+        draftTransaction?.asset.details?.symbol
+      }`;
+    }
+
+    const ethTransactionTotalMaxAmount = Number(
+      hexWEIToDecETH(hexMaximumTransactionFee),
+    );
+
+    const primaryTotalTextOverrideMaxAmount = `${title} + ${ethTransactionTotalMaxAmount} ${nativeCurrency}`;
+
+    const detailText = (
+      <Box
+        height={BLOCK_SIZES.MAX}
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.COLUMN}
+        className="gas-display__total-value"
+      >
+        <LoadingHeartBeat estimateUsed={draftTransaction?.userFeeLevel} />
+        <UserPreferencedCurrencyDisplay
+          type={SECONDARY}
+          key="total-detail-text"
+          value={hexTransactionTotal}
+          hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
+        />
+      </Box>
+    );
```

Comment:
I would prefer to just see this inline in the JSX then to have to scroll around to see what it is. In general the only time I think we should advise to set a variable to a JSX element is when that element will be manipulated (ie if it was let detailText and we were modifying what component to render based on logic). For example, detailTotal and maxAmount are actually perfectly fine to do this way because they are being modified and it makes the code more readable to have that logic clearly broken apart out of the main JSX statement.

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/c64948a3c0494308f824cdc443a9b04b010b9054).